src/catkin_pkg/packages.py

```diff
@@ -32,13 +32,24 @@
 
 """Library to find packages in the filesystem."""
 
-import multiprocessing
 import os
 
 from .package import _get_package_xml
 from .package import PACKAGE_MANIFEST_FILENAME
 from .package import parse_package_string
 
+if os.name == 'nt':
+    # https://docs.python.org/2/library/multiprocessing.html#windows
+    #
+    # On Windows, using multiprocessing requires the scripts to be
+    # "Safe importing of main module" to avoid RuntimeError.
+    # This restriction requires downstream tools to be modified to
+    # work. Instead, let's fallback to threading wrapper to get
+    # around it and still keep the parallelism on Windows.
+    import multiprocessing.dummy as multiprocessing
```

Comment:
What is the impact of using threading for this on Windows? Is the performance still better than without?

Reply:
Well... ran some benchmark tests today and in a test case where it has 1000 packages (and each has dummy 1000 <build_depend> nodes), the result looks not good for threading wrapper in general on Windows...

On CPython 2.7:
- Disable parallel: 26 sec
- Using multiprocessing: 7.5 sec
- Using multiprocessing.dummy: 98.5 sec

On Anaconda Python 3.7:
- Disable parallel: 12.8 sec
- Using multiprocessing: 4.8 sec
- Using multiprocessing.dummy: 14.5 sec

(My environment is using `Intel® Xeon® Processor E5-1650 v4` 6-cores CPU.)
(My test case is `test_find_packages_with_large_amount_packages` in https://github.com/seanyen-msft/catkin_pkg/blob/windows_port_multiprocessing_disabled/test/test_packages.py)

Reply:
The results don't look like this should be merged as is

Reply:
That make senses. Since now we explored some approaches, any pointers what to investigate next at this moment?